src/TrackGenerator.cpp

```diff
@@ -469,22 +471,22 @@ void TrackGenerator::retrieveTrackCoords(double* coords, int num_tracks) {
  *
  * @code
  *          num_segments = track_generator.getNumSegments()
- *          coords = track_generator.retrieveSegmentCoords(num_segments*5)
+ *          coords = track_generator.retrieveSegmentCoords(num_segments*7)
  * @endcode
  *
- * @param coords an array of coords of length 5 times the number of segments
+ * @param coords an array of coords of length 7 times the number of segments
  * @param num_segments the total number of Track segments
  */
 void TrackGenerator::retrieveSegmentCoords(double* coords, int num_segments) {
 
-  if (num_segments != 5*getNumSegments())
+  if (num_segments != 7*getNumSegments())
     log_printf(ERROR, "Unable to retrieve the Track segment coordinates since "
                "the TrackGenerator contains %d segments with %d coordinates "
                "but an array of length %d was input",
-               getNumSegments(), 5*getNumSegments(), num_segments);
+               getNumSegments(), 7*getNumSegments(), num_segments);
 
   segment* curr_segment = NULL;
-  double x0, x1, y0, y1;
+  double x0, x1, y0, y1, z;
```

Comment:
It looks to me like there is an inconsistency here. I understand that you just have ``z`` instead of ``z0`` and ``z1`` because it is just a 2D slide at a given z-height, but you are returning an array that has two extra variable slots (you changed from 5 to 7). Perhaps we should just include ``z0`` and ``z1`` to make this consistent and make the transition to 3D a bit smoother.